ban/commands/ignsna.py

```diff
@@ -0,0 +1,48 @@
+import os
+from ban.commands import command, report
+from ban.core.models import (Municipality, PostCode)
+from .helpers import session, batch, nodiff
+
+__namespace__ = 'import'
+
+
+@command
+@nodiff
+def ignsna(path, **kwargs):
+    """Import from IGN/Laposte BDUNI
+    :param path: directory location of the files hsp7aaaa.ai, hsv7aaaa.ai and hsw4aaaa.ai"""
+
+    municipality_zipcode_file = os.path.join(path, 'hsp7aaaa.ai')
+    if municipality_zipcode_file is not None:
+        lines = open(municipality_zipcode_file).readlines()
+        batch(process_municipality_file, lines, max_value=len(lines))
+
+
+@session
+def process_municipality_file(line):
+    if line[50] == 'M':
```

Comment:
Trick: instead of doing `if line[50] == 'M'`, do:
```python
if line[50] != 'M':
    return  # BTW, if it's not expected, report it ;)
# my code continue here without useless indentation
```